templates/service.yaml

```
@@ -1,7 +1,11 @@
 apiVersion: v1
 kind: Service
 metadata:
+{{- if .Values.service.name }}
+  name: {{ .Values.service.name }}
+{{- else }}
```

Alternatively, I think this could be written as:
```suggestion
  name: {{ .Values.service.name | default template "docker-registry.fullname" . }}
```